packages/js/product-editor/src/hooks/use-product-manager/use-product-manager.ts

```diff
@@ -70,17 +70,21 @@ export function useProductManager< T = Product >( postType: string ) {
 			setIsSaving( true );
 
 			await validate( extraProps );
+			const { saveEntityRecord } = dispatch( 'core' );
 
-			// @ts-expect-error There are no types for this.
-			const { editEntityRecord, saveEditedEntityRecord } =
-				dispatch( 'core' );
-
-			await editEntityRecord< T >( 'postType', postType, id, extraProps );
+			const { blocks, content, selection, ...editedProduct } = wpSelect(
+				'core'
+				// @ts-expect-error There are no types for this.
+			).getEditedEntityRecord( 'postType', postType, id );
 
-			const savedProduct = await saveEditedEntityRecord< T >(
+			const savedProduct = await saveEntityRecord(
 				'postType',
 				postType,
-				id,
+				{
+					...editedProduct,
+					...extraProps,
+				},
+				// @ts-expect-error There are no types for this.
```

Comment:
@octaedro @mdperez86 as just a follow up, given we had to create a CFE for this code. Whenever we update requests logic its important to check the request payload before the change and after the change.

As we would of noticed that before this change only the edited changes would of been included in the payload and after this change all product data.

Something that is also valuable if we change GB functions in the future, is to look at the original one. For example, you can see that `saveEditedEntityRecords` [used `getEntityRecordNonTransientEdits`](https://github.com/WordPress/gutenberg/blob/6f2e542eca78f846b80a475d8ba250f82fa81f30/packages/core-data/src/actions.js#L792) and not `getEditedEntityRecord`.